src/main/java/duke/subfiles/TaskList.java

```diff
@@ -0,0 +1,396 @@
+package duke.subfiles;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+import duke.exceptions.DateFormatException;
+import duke.exceptions.EmptyDateException;
+import duke.exceptions.EmptyDescriptionException;
+import duke.exceptions.InvalidInputException;
+import duke.exceptions.ListOutOfBoundsException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+/**
+ * The TaskList class contains a list of tasks created by
+ * user input, and allows the user to add, print, or delete
+ * tasks, as well as to mark a task in the list as done.
+ *
+ * @author  arsatis
+ * @version 1.1
+ * @since   2021-01-26
+ */
+public class TaskList {
+    /** List of tasks created by user input. */
+    private ArrayList<Task> tasks;
+
+    /**
+     * Default constructor for the TaskList class.
+     */
+    public TaskList() {
+        tasks = new ArrayList<>();
+    }
+
+    /**
+     * Adds a to-do to the list of tasks.
+     *
+     * @param s User input triggering the addition of a to-do
+     *          to the list of tasks.
+     * @throws EmptyDescriptionException If no description is provided
+     *                                   for the to-do.
+     */
+    private void addTodo(String s)

```

Comment:
Can consider changing parameter name to something more meaningful, e.g. "description" 😄